web/Controllers/LandlordController.cs

```diff
@@ -44,25 +48,37 @@ public ActionResult RegisterGet()
     public async Task<ActionResult> RegisterPost([FromForm] LandlordProfileModel createModel)
     {
         // This does checks based on the annotations (e.g. [Required]) on LandlordProfileModel
-        if (!ModelState.IsValid)
-        {
-            return View("Register");
-        }
+        if (!ModelState.IsValid) return View("Register");
 
         var user = GetCurrentUser();
+        ILandlordService.LandlordRegistrationResult result;
+        int? id = null;
+
+        if (createModel.Unassigned && user.IsAdmin)
+        {
+            var resultTuple = await _landlordService.RegisterLandlord(createModel);
+            result = resultTuple.result;
+            id = resultTuple.id;
+        }
+        else
+        {
+            result = await _landlordService.RegisterLandlord(createModel, user);
+        }
 
-        var result = await _landlordService.RegisterLandlordWithUser(createModel, user);
 
         switch (result)
         {
             case ILandlordService.LandlordRegistrationResult.Success:
-                _logger.LogInformation("Successfully created landlord for user {UserId}", user.Id);
-                return Redirect(Url.Action("MyProfile")!);
+                _logger.LogInformation("Successfully created landlord for user");
+                if (!createModel.Unassigned)
+                    return Redirect(Url.Action("MyProfile")!);
+                //return RedirectToAction("LandlordList", "Admin");

```

Comment:
Can delete this comment.

It's also easy to make mistakes doing `if` without braces. It's fine for things in one line like you've done below (e.g. `if (landlord == null) return StatusCode(404);`), but I'd use braces here.

Reply:
Apologies, I've set up an automatic code format thing in rider and its a little overzealous. It has stopped all the "extra whitespace/newline" here comments though! ^^'

Reply:
Fixed it :) told it to keep embedded blocks as they are in the code